src/main/java/org/folio/spring/filter/LoggingRequestFilter.java

```diff
@@ -0,0 +1,125 @@
+package org.folio.spring.filter;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.time.Instant;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.GenericFilterBean;
+import org.springframework.web.util.ContentCachingRequestWrapper;
+import org.springframework.web.util.ContentCachingResponseWrapper;
+
+import org.folio.spring.integration.XOkapiHeaders;
+
+@Log4j2
+@Component
+@ConditionalOnProperty(
+  prefix = "folio.logging.request",
+  name = "enabled",
+  havingValue = "true"
+)
+public class LoggingRequestFilter extends GenericFilterBean {
+
+  private static final String START_TIME_ATTR = "startTime";
+
+  private final Level level;
+
+  public LoggingRequestFilter(@Value("${folio.logging.request.level: FULL}") Level level) {
+    this.level = level;
+  }
+
+  @Override
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+    throws IOException, ServletException {
+    if (log.isInfoEnabled()) {
+      filterWrapped(wrapRequest(request), wrapResponse(response), chain);
+    } else {
+      chain.doFilter(request, response);
+    }
+  }
+
+  private void filterWrapped(ContentCachingRequestWrapper request, ContentCachingResponseWrapper response,
+                             FilterChain chain) throws ServletException, IOException {
+    filterBefore(request);
+    chain.doFilter(request, response);
+    filterAfter(request, response);
+    response.copyBodyToResponse();
+  }
+
+  private void filterBefore(ContentCachingRequestWrapper request) throws UnsupportedEncodingException {
+    request.setAttribute(START_TIME_ATTR, Instant.now().toEpochMilli());
+
+    var requestId = getRequestId(request);
+
+    log.info("[{}] ---> {} {} {}",
```

Comment:
In my opinion it is too verbose. Such logging would litter logs, and guys would just disable it.
One line with okapi tenant, okapi_user (if applicable) and request_id will be enough.

Reply:
It could be similar to the pattern in logs configuration:
appender.console.layout.pattern = %d{HH:mm:ss} [$${folio:requestid}] [$${folio:tenantid}] [$${folio:userid}] [$${folio:moduleid}] %-5p %-20.20C{1} %m%n

Reply:
@MikhailFokanov, thank you for your comment. 
You could look at examples of log messages in the README file. As you can see, they are in folio format. 
As for a more detailed log, I think it could be helpful.